### PR TITLE
docs: add troubleshooting guide for AADSTS90061

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Add troubleshooting guide for `AADSTS90061`

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes https://github.com/Azure/azure-workload-identity/issues/605

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:

### CODE REVIEW

File: docs/book/src/troubleshooting.md [hunk 1]
```diff
@@ -58,3 +58,32 @@ If you encounter the error above, it means that the issuer of the service accoun
...
+   at Azure.Identity.ClientAssertionCredential.GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        StatusCode: 400
+        ResponseBody: {"error":"invalid_request","error_description":"AADSTS90061: Request to External OIDC endpoint failed.\r\nTrace ID: 86c81de3-efbd-4c0f-854d-cb8f329e5b00\r\nCorrelation ID: a92568a5-cc37-4136-ad4d-1ba82988abc2\r\nTimestamp: 2022-10-21 04:36:48Z","error_codes":[90061],"timestamp":"2022-10-21 04:36:48Z","trace_id":"86c81de3-efbd-4c0f-854d-cb8f329e5b00","correlation_id":"a92568a5-cc37-4136-ad4d-1ba82988abc2"}
+...
+```
+
+If you encounter the error above, it means the OIDC issuer endpoint is not exposed to the internet or is inaccessible. You can verify this by running the following commands to check if the endpoints are accessible:
+
+```bash
+curl ${ISSUER_URL}/.well-known/openid-configuration
```
Comment (sozercan) [inline docs/book/src/troubleshooting.md:87]: what's `ISSUER_URL` here?

Comment (aramase) [reply docs/book/src/troubleshooting.md]: Added a note!

**Review by sozercan — APPROVED** [review]

**Comment by mblaschke** [comment]
@aramase
I've just encountered this issue on multiple managed AKS clusters but this "troubleshooting guide" for AADSTS90061 doesn't really help.

Is there any additional documentation available what to do when AKS is failing to publish this endpoint? Or is that a issue where users have to create an Azure support ticket?

**Comment by aramase** [comment]
> I've just encountered this issue on multiple managed AKS clusters but this "troubleshooting guide" for AADSTS90061 doesn't really help.

@mblaschke please open a support ticket. If the endpoint is not accessible, then this could be an AKS issue. If the endpoint is exposed and accessible, then AAD team can take a look.

**Pull request merged**